image-file.c

```diff
@@ -29,6 +29,36 @@ struct file {
 	cfg_bool_t copy;
 };
 
+static int file_parse_holes(struct image *image, cfg_t *cfg)
+{
+	int i;
+
+	image->n_holes = cfg ? cfg_size(cfg, "holes") : 0;
+	if (image->n_holes == 0)
+		return 0;
+
+	image->holes = xzalloc(image->n_holes * sizeof(*image->holes));
+	for (i = 0; i < image->n_holes; i++) {
+		const char *s = cfg_getnstr(cfg, "holes", i);
+		char *start, *end;
+		int len;
+
+		if (sscanf(s, " ( %m[0-9skKMG] ; %m[0-9skKMG] ) %n", &start, &end, &len) != 2 ||
+		    len != (int)strlen(s)) {
+			image_error(image, "invalid hole specification '%s', use '(<start>;<end>)'\n",
+				    s);
+			return -EINVAL;
+		}
+
+		image->holes[i].start = strtoul_suffix(start, NULL, NULL);
+		image->holes[i].end = strtoul_suffix(end, NULL, NULL);
+		free(start);
+		free(end);
+		fprintf(stderr, "added hole (%llu, %llu)\n", image->holes[i].start, image->holes[i].end);
```

Comment:
``` suggestion
		image_debug(image, "added hole (%llu, %llu)\n", image->holes[i].start, image->holes[i].end);
```

Reply:
Gah, yeah, that was really a debug printf leftover that I meant to remove. But I guess it does make sense to keep it as an image_debug. Thanks.